tahsinhasem iP

tahsinhasem opened the pull request:
# Dude :muscle:
A ~~easy~~ kinda easy to use cli task manager.

### Features
- add todos, deadlines, events
- remove events
- search for tasks!

### Why should you use it?
1. fast
2. easy to learn
3. useful help documentation!!

Its as **easy** as:
```java
    public static void main(String[] args) {
        new Dude("data/tasklist.ser").run();
    }
```
where `Dude` is the main class of this program.

### Advanced Features
- [X] Store past tasks
- [X] Find tasks by keyword search
-  [ ] Connected to ChatGPT [OpenAI](https://openai.com)

> "I am only making this cuz I want a degree" - said no man ever, **right?**  :shrug:

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

tah5in committed:
Renamed duke to dude, added greet and bye

tah5in committed:
Added echo feature, added indentation to responses

tah5in committed:
Added TaskList.java, Implemented Adding Tasks

tah5in committed:
Changed Logo

tah5in committed:
Added Task.java

tah5in committed:
Refactored Task List to work with Task objects instead of Strings

tah5in committed:
Refactored TaskList to throw errors

tah5in committed:
Changed TaskListFullException to be public

tah5in committed:
Renamed variable s to msg

tah5in committed:
Added mark_ as done feature

tah5in committed:
Changed the output message to include task number for marking and unmarking tasks

tah5in committed:
Added Unmark Feature

tah5in committed:
Added handling case for empty input - "".

tah5in committed:
Added Todo class

tah5in committed:
Added Deadline class

tah5in committed:
Added Event class

tah5in committed:
Added a utils class containing utility function(s)

tah5in committed:
Implemented a factory function for Todo

tah5in committed:
Implemented todo

tah5in committed:
Implemented deadline command

tah5in committed:
Changed task add message

tah5in committed:
Added deadline command

tah5in committed:
Added deadline command

tah5in committed:
Bug fixes

tah5in committed:
Added DukeException.java and subclasses for error handling

tah5in committed:
Renamed DukeException to DudeException

tah5in committed:
Added InvalidFormatException.java

tah5in committed:
Added appropriate exceptions to Event Class

tah5in committed:
Added appropriate exceptions to Todo Class

tah5in committed:
Added appropriate exceptions to Deadline Class

tah5in committed:
Refactored to proper error handling

tah5in committed:
Removed default file

tah5in committed:
Removed default from switch statement

tah5in committed:
Merge branch 'master' of https://github.com/tahsinhasem/ip

tah5in committed:
Refactored list from array to ArrayList<Task>

tah5in committed:
Removed use of count to keep track of tasks. Replaced with arraylist size

tah5in committed:
Handle error for non integer ids

tah5in committed:
Added testing for delete command

Ty-stan0417 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,63 @@
+        this.deadline_date = by;
+    }
+
+    @Override
+    public void markAsDone() {
+        super.markAsDone();
+    }
+
+    @Override
+    public void markAsUndone() {
+        super.markAsUndone();
+    }
```
Consider to delete the lines of code, as Deadline have same behaviour with Task when mark or unmark it.

Ty-stan0417 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,63 @@
+        super.markAsUndone();
+    }
+
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + " (by: " + deadline_date + ")";
+    }
```
Good use of Overriding.

Ty-stan0417 commented on src/main/java/Dude.java:
```diff
@@ -0,0 +1,204 @@
+import java.util.NoSuchElementException;
+import java.util.Scanner;
+
+public class  Dude {
```
`public class  Dude` In your code, there are 2 spaces between class and Dude. Consider to follow the coding standard that this is a space exactly.

Ty-stan0417 commented on src/main/java/Dude.java:
```diff
@@ -0,0 +1,204 @@
+
+public class  Dude {
+
+    static TaskList taskList = new TaskList();
```
Consider to add the access modifier to the taskList as the variable are suggested code with `private` or `protected`.

Ty-stan0417 commented on src/main/java/Dude.java:
```diff
@@ -0,0 +1,204 @@
+                continue;
+            }
+
+            switch (command){
+                case "bye":
+                    System.out.println(bye());
+                    return;
+                case "list":
+                    System.out.println(list());
+                    break;
+                case "mark":
+                    System.out.println(mark_as_done(msg));
+                    break;
+                case "unmark":
+                    System.out.println(mark_as_undone(msg));
+                    break;
+                case "todo":
+                    System.out.println(handle_todo_command(msg));
+                    break;
+                case "event":
+                    System.out.println(handle_event_command(msg));
+                    break;
+                case "deadline":
+                    System.out.println(handle_deadline_command(msg));
+                    break;
+                case "delete":
+                    System.out.println(handle_delete_command(msg));
+                    break;
+            }
```
LGTM. I like the way how you are handling your command. The code looks short and easy to understand.

Ty-stan0417 commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,77 @@
+public class Event extends Task{
+
+    private final String from_time;
+    private final String to_time;
```
Try to store the `from_time` and `to_time` with LocalDate, this will be done in Level-8.

Ty-stan0417 commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,72 @@
+
+    public String add_task(Task task) throws TaskListFullException {
+
+        if ( list.size() >= 100) {
+            throw new TaskListFullException("Sorry, the task list is full.");
+        }
```
Consider to remove this as you are using the `ArrayList<Task>`. This can allow user to store more task in the list.

Ty-stan0417 commented on src/main/java/Todo.java:
```diff
@@ -0,0 +1,34 @@
+        super(description);
+    }
+
+    @Override
+    public void markAsDone() {
+        super.markAsDone();
+    }
+
+    @Override
+    public void markAsUndone() {
+        super.markAsUndone();
+    }
```
Consider to remove these lines as they have same behaviour with the superclass of them.

Ty-stan0417 reviewed (commented):
Overall, LGTM. Good job!👌
Careful when using the overriding, when a subclass have same behaviour of its superclass, you don't need to override it.

tah5in committed:
Added Storage Class

tah5in committed:
Added create storage function

tah5in committed:
Added Deleting to Storage Class

tah5in committed:
Refactored into Packages

tah5in committed:
Added saving taskList functiuonality

tah5in committed:
Fixed edge case bug where exception would be thrown if there was an empty tasklist.ser file

tah5in committed:
Improved readability

tah5in committed:
Implemented Auto save

tah5in committed:
Merge branch 'branch-Level-7'

tah5in committed:
Implemented ENUMs for commands

tah5in committed:
Added parseDate and fotmatDate functions to Task class

tah5in committed:
Implemented parsing datetimes for Deadlines

tah5in committed:
implemented date formatting for deadlines

tah5in committed:
Implemented DateTime parsing for Events

tah5in committed:
Implemented formatting of dates for Events

tah5in committed:
Changed printing of time format to `MMM d, yyyy @ h:mma`

tah5in committed:
Merge branch 'branch-Level-8'

tah5in committed:
Changed Storage to take in save file location. Make it more OOP

tah5in committed:
removed main method from storage

tah5in committed:
added skeleton for command package

tah5in committed:
Added Commands

tah5in committed:
Added Command Type Enums

tah5in committed:
Changed getTask to take in ID instead of index

tah5in committed:
Dixed bug

tah5in committed:
Working duke in static context

tah5in committed:
Better code quality

tah5in committed:
Merge branch 'A-OOP'

tah5in committed:
renamed CommandParser to Parser. Moved CommandTypes to Command Package

tah5in committed:
Merge branch 'A-OOP'

tah5in committed:
moved all classes to dude package

tah5in committed:
Merge branch 'add-gradle-support'

tah5in committed:
updated build.gradle

tah5in committed:
Added Checksyle. Changed Style for Dude.java

tah5in committed:
Added tests for Storage and Parser

tah5in committed:
Added .equals for Todo

tah5in committed:
Added Test cases for Todo Task

tah5in committed:
Added Test cases for Deadline

tah5in committed:
Added equals method

tah5in committed:
changed method spelling mistake for deadline test

tah5in committed:
Added test cases for event class

tah5in committed:
Fixed main class not found bug. Fixed shadowJar error

tah5in committed:
Added JavaDoc for Dude and Storage Classes

tah5in committed:
added JavaDoc and coding style changes

tah5in committed:
Fixed ordering of import statements following coding standard

tah5in committed:
Added Javadoc to the Tasks package

tah5in committed:
Added Javadoc to the Exceptions package

tah5in committed:
Added Javadoc to the Commands package

tah5in committed:
Implemented find function

tah5in committed:
Implemented find command

tah5in committed:
Added find command to Parser

tah5in committed:
Added help support for find command

tah5in committed:
Bug fixes for UI

tah5in committed:
Coding Style fixes for Deadline Class

tah5in committed:
Merge branch 'branch-level-9'

tah5in committed:
Coding Style fixes for Event Class

tah5in committed:
Added Javadoc for Task class

tah5in committed:
Updated Event, TaskList, Todo classes to coding standards

tah5in committed:
Updated code to coding standards

tah5in committed:
Renamed Tasks package to tasks

tah5in committed:
Renamed all packages to lower-case

tah5in committed:
Updated to coding Standards

tah5in committed:
Updated to coding Standards

tah5in committed:
Changed array indenting for HelpCommand

tah5in committed:
Changed array indenting for HelpCommand

tah5in committed:
changed indent for Help copmmand

tah5in committed:
Added Javadoc to utils class

tah5in committed:
Added Javadoc to utils class

tah5in committed:
Renamed utils class to Utils

tah5in committed:
Updated TaskListFullException to coding standards

tah5in committed:
Updated MarkCommand to coding standards

tah5in committed:
Code quality updates from intelliJ suggestions

tah5in committed:
Minor style change

tah5in committed:
Fixed Merge Conflicts

tah5in committed:
Fixed some Checkstyle errors

tah5in committed:
Implemented sending message to UI

tah5in committed:
Implemented exchange of dialog

tah5in committed:
minor changes

tah5in committed:
Added proper images

tah5in committed:
Added ActionTo Text Input Field, Changed color for labels

tah5in committed:
Changed to string to return "No tasks in the list! :(" when empty

tah5in committed:
Fixed bug where label was not showing entire text

tah5in committed:
Changed DialogBox.fxml so that the text has some margin

tah5in committed:
Merge branch 'branch-Level-10'

tah5in committed:
bug fix

tah5in committed:
Merge branch 'branch-Level-10'

tah5in committed:
cleaned imports

tah5in committed:
Implmented var-args in TaskList

tah5in committed:
Added padding to dialog container

tah5in committed:
Added Assertions to Commands.

There were no assertions in the Commands that were being implemented in
dude. Especially the constructors, where many of the constructions were
assuming certain conditions to hold true.

Conditions that the command class sub-class constructors assumed to
be true:
* String input : Assumed to be not null, assumed to contain appropriate
command
* Tasklist tasklist: Assumed to be not null

Now assertions have been added to the constructors to make sure such
conditions are actually holding true.

These assertions will help to reduce bugs making sure the conditions
assumed to be true, actually hold true.

I believe this will drastically reduce bugs when I decide to add on more
features.

tah5in committed:
Added Assertions to Commands.

There were no assertions in the Commands that were being implemented in
dude. Especially the constructors, where many of the constructions were
assuming certain conditions to hold true.

Conditions that the command class sub-class constructors assumed to
be true:
* String input : Assumed to be not null, assumed to contain appropriate
command
* Tasklist tasklist: Assumed to be not null

Now assertions have been added to the constructors to make sure such
conditions are actually holding true.

These assertions will help to reduce bugs making sure the conditions
assumed to be true, actually hold true.

I believe this will drastically reduce bugs when I decide to add on more
features.

tah5in committed:
added assertions to constructors of tasks packages

tah5in committed:
added assertions to constructors of tasks packages

tah5in committed:
Added assertions to Storage.java public functions

tah5in committed:
Added assertions to Storage.java public functions

tah5in committed:
Added assertion to dsicardFirstWord method, modified javadoc for findIndex method.

tah5in committed:
Added assertion to dsicardFirstWord method, modified javadoc for findIndex method.

tah5in committed:
Changed the main class in gradle

tah5in committed:
Refactored the GUI Views inside a GUI package

tah5in committed:
Extracted loadMainScene method out.

tah5in committed:
Added javadoc to Launcher.java

tah5in committed:
Refactored to remove unnecessary methods from Dude.javb

tah5in committed:
Added more javadoc to dude.java

tah5in committed:
Code style improvements for Dude.java

tah5in committed:
Added javaDoc for main.start()

tah5in committed:
better code quality for Task package

tah5in committed:
better code quality for Task package

tah5in committed:
better code quality for Task package

tah5in committed:
better code quality for Task package

tah5in committed:
fixed some checkstyle errors

tah5in committed:
fixed some checkstyle errors

tah5in committed:
Better code quality

tah5in committed:
Better code quality

tah5in committed:
Better code quality

tah5in committed:
Better code quality

tahsinhasem committed:
Merging branch a assertions 

adding assertions to main code

tahsinhasem committed:
Merge branch 'branch-A-CodeQuality' into master

tahsinhasem committed:
Merge Main into branch-A-CodeQuality (#3)

tahsinhasem committed:
Merge branch-A-CodeQuality to master (#4)

tah5in committed:
handled merge conflicts due to previous rebase

tah5in committed:
Changed Task::from methods to use Task:add_task method

tah5in committed:
Added Sorting functionality to Dude.

Currently, tasks are stored such that they are ordered by when tasks are
added into dude by user. It has no order.

The ordering of the tasks are necessary as it is a relevant information
to the user.

I have modified Dude in such a way that it sorts the tasks added by
time, every time a new task is added to added it dude. The Tasks are
sorted by By-Dates (if Deadline Task) or From-Date (for event tasks).
Todo tasks are sorted by which one was last added.

Now the user will get a sorted task list everywhere in the program,
allowing the user to see tasks chronologically.

tah5in committed:
Added testing for sorting tasks

tah5in committed:
Removed white line from GUI

tah5in committed:
Added showing welcome message

tah5in committed:
Tried to fix bug where dialog is truncated

tah5in committed:
Added Documentation

tah5in committed:
Fixed bug where text was getting truncated
